dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -0,0 +1,147 @@
+[[getting-traffic-into-cluster]]
+= Getting Traffic Into The Cluster
+{product-author}
+{product-version]
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+ifdef::openshift-origin,openshift-enterprise,openshift-dedicated[]
+== Overview
+There are many ways to access the cluster. This section describes some
+commonly used approaches.
+
+The recommendation is:
+
+- If you have HTTP/HTTPS -- Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
+- If you have Encrypted TCP (i.e., TLS with the SNI header) -- Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
+- If you have TCP or UDP -- Use xref:../dev_guide/getting_traffic_into_cluster.adoc#using-the-loadbalancer[LoadBalancer],
```

Comment:
On 09/13/2016 10:41 AM, Ben Bennett wrote:
>
> In dev_guide/getting_traffic_into_cluster.adoc 
> <https://github.com/openshift/openshift-docs/pull/2827#discussion_r78573259>:
>
> > +:toc: macro
> > +:toc-title:
> > +:prewrap!:
> > +
> > +toc::[]
> > +
> > +ifdef::openshift-origin,openshift-enterprise,openshift-dedicated[]
> > +== Overview
> > +There are many ways to access the cluster. This section describes some
> > +commonly used approaches.
> > +
> > +The recommendation is:
> > +
> > +- If you have HTTP/HTTPS -- Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
> > +- If you have Encrypted TCP (i.e., TLS with the SNI header) -- Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
> > +- If you have TCP or UDP -- Use xref:../dev_guide/getting_traffic_into_cluster.adoc#using-the-loadbalancer[LoadBalancer],
>
> This is the previous comment:
>
> Perhaps we can make a table comparing features now we have 5 options. 
> Or a flow chart...
>
>   * If you have HTTP/HTTPS -- Use the router
>   * If you have Encrypted TCP (i.e. TLS with the SNI header) -- Use
>     the router
>   * If you have TCP or UDP
>       o Use the new bare-metal LoadBalancer, but that limits you to a
>         single ingress IP (can be a VIP, but still will be a single
>         machine for initial load balancing). It simplifies the admin
>         job, but uses one IP per service.
>       o Manually assign ExternalIPs to the service. Here you can
>         assign a set of IPs, so can have multiple machines for the
>         incoming load balancing. But requires elevated permissions to
>         assign, and manual tracking of what IP:ports have been used.
>       o Use NodePorts to expose the service on ALL nodes in the
>         cluster. More wasteful of scarce port resources. BUT a little
>         easier to set up multiple. Again, requires more privilege.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/openshift/openshift-docs/pull/2827/files/70b1b45900d11983523680b7dd84ece06d6c5253#r78573259>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ANUgeineLA4bihP0eXR02G7tvJ6kuzz6ks5qprYDgaJpZM4J6qBl>.
>
thanks, I lost it in the stream ...